src/components/picker/helpers/usePickerSelection.tsx

```diff
@@ -20,7 +21,11 @@ const usePickerSelection = (props: UsePickerSelectionProps) => {
     }
   }, [value]);
 
-  const onDoneSelecting = useCallback((item: PickerValue) => {
+  const onDoneSelecting = useCallback(async (item: PickerValue) => {
+    // Using !(await onItemPress?.(item)) does not work properly when onItemPress is not sent
+    if (onItemPress && !(await onItemPress(item))) {
```

Comment:
We already have an onPress callback in PickerItemProps, do you think we should use that?

Reply:
The other way I found to do this is shown in [another PR](https://github.com/wix/react-native-ui-lib/pull/2787), you choose